src/drawing/scrolling_text.c

```diff
@@ -475,10 +1547,10 @@ void scrolling_text_set_bitmap_for_ttf(utf8 *text, int scroll, uint8 *bitmap, co
 	while (true) {
 		// Skip any none displayed columns
 		if (scroll == 0) {
-			sint16 scrollPosition = *scrollPositionOffsets;
-			if (scrollPosition == -1) return;
-			if (scrollPosition > -1) {
-				uint8 *dst = &bitmap[scrollPosition];
+			rct_xy8 scrollPosition = *scrollPositionOffsets;
+			if (scrollPosition.xy == (uint16)-1) return;
+			if ((sint16)scrollPosition.xy > -1) {
+				uint8 *dst = &bitmap[scrollPosition.x + scrollPosition.y * 64];
```

Comment:
While the variables are stored nicely now, the drawback is multiplication which has to be done here. But perhaps compiler can figure it out and optimise or use the `lea` for calculating the address here.